hikari/impl/rest.py

```diff
@@ -1270,49 +1276,64 @@ async def _create_message(
         final_attachments: typing.List[files.Resource[files.AsyncReader]] = []
         if attachment is not undefined.UNDEFINED:
             final_attachments.append(files.ensure_resource(attachment))
-        if attachments is not undefined.UNDEFINED:
+        elif attachments is not undefined.UNDEFINED:
```

Comment:
btw changing this to an elif is kinda unnecessary since an earlier check already asserts that only one is not undefined